internal/provider/resource_model_test.go

```diff
@@ -146,6 +146,77 @@ func TestAcc_ResourceModel_UpgradeProvider(t *testing.T) {
 	})
 }
 
+func TestAcc_ResourceModel_Annotations_Basic(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-model")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAnnotationsModel(modelName, "test"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_model.testmodel", "name", modelName),
+					resource.TestCheckResourceAttr("juju_model.testmodel", "annotations.test", "test"),
+				),
+			},
+			{
+				Config: testAccAnnotationsModel(modelName, "test-update"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_model.testmodel", "name", modelName),
+					resource.TestCheckResourceAttr("juju_model.testmodel", "annotations.test", "test-update"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
+resource "juju_model" "testmodel" {
+  name = %q
+}`, modelName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_model.testmodel", "name", modelName),
+					resource.TestCheckNoResourceAttr("juju_model.testmodel", "annotations.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_ResourceModel_Annotations_DisjointedSet(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-model")
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+resource "juju_model" "testmodel" {
+  name = %q
+   annotations = {
+	test = "test"
+  }
+}`, modelName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_model.testmodel", "name", modelName),
+					resource.TestCheckResourceAttr("juju_model.testmodel", "annotations.test", "test"),
+				),
+			},
+			{
+				Config: fmt.Sprintf(`
```

Comment:
todo: use `testAccAnnotationsModel()` here.

Update `testAccAnnotationsModel()` to pass in  `key` and `value` args for the annotation.